<a href="https://colab.research.google.com/github/lipeoak/BeeCrowd__Problems/blob/main/beecrowd_exercises.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>



# **Exercises approved by the testing program of BeeCrowd**
#**In case any of the scripts are not presented in github, feel free to open it directly on google colab notebook**

#**Link:** https://colab.research.google.com/drive/1PVlQslHzVn_eV5wvbgcbd8-wZdfKrxBX#scrollTo=7TkvgVtwWE-4

#**My LinkedIn Profile:** https://www.linkedin.com/in/felipemarcolinocarvalhocosta/



#Begginer

#####BEE 1010 - Simple Calculate


In this problem, the task is to read a code of a product 1, the number of units of product 1, the price for one unit of product 1, the code of a product 2, the number of units of product 2 and the price for one unit of product 2. After this, calculate and show the amount to be paid.

**Input**
The input file contains two lines of data. In each line there will be 3 values: two integers and a floating value with 2 digits after the decimal point.

**Output**
The output file must be a message like the following example where "Valor a pagar" means Value to Pay. Remember the space after ":" and after "R$" symbol. The value must be presented with 2 digits after the point.

|Input Samples|Output Samples|
|-------------|--------------|
|12 1 5.30|VALOR A PAGAR R\$ 15.50|
|16 2 5.20||
|||
|13 2 15.30|VALOR A PAGAR R\$ 51.40|
|161 4 5.20||
|||
|1 1 15.10|VALOR A PAGAR R\$ 30.20|
|2 1 15.10||

In [ ]:
product_1 = input().split()
product_1[0] = int(product_1[0])
product_1[1] = int(product_1[1])
product_1[2] = float(product_1[2])

product_2 = input().split()
product_2[0] = int(product_2[0])
product_2[1] = int(product_2[1])
product_2[2] = float(product_2[2])

total = (product_1[1]*product_1[2])+(product_2[1]*product_2[2])
print(f'VALOR A PAGAR: R$ {total:.2f}')

1 1 15.10
2 1 15.10
VALOR A PAGAR: R$ 30.20


#####BEE 1014 - Consumption

Calculate a car's average consumption being provided the total distance traveled (in Km) and the spent fuel total (in liters).

**Input**
The input file contains two values: one integer value X representing the total distance (in Km) and the second one is a floating point number Y  representing the spent fuel total, with a digit after the decimal point.

**Output**
Present a value that represents the average consumption of a car with 3 digits after the decimal point, followed by the message "km/l".

|Input Sample|Output Sample|
|-|-|
|500|14.286 km/l|
|35.0||
|||
|2254|18.119 km/l|
|124.4||
|||
|4554|9.802 km/l|
|464.6||



In [ ]:
distance = int(input())
fuel_spent = float(input())
consumption = (distance/fuel_spent)

print(f'{consumption:.3f} km/l')

2254
124.4
18.119 km/l


#####BEE 1015 - Distance Between Two Points

Read the four values corresponding to the x and y axes of two points in the plane, p1 (x1, y1) and p2 (x2, y2) and calculate the distance between them, showing four decimal places, according to the formula:

Distance = $$ \sqrt{(x_2 - x_1)^2 + (y_2 - y_1)^2} $$

**Input**
The input file contains two lines of data. The first one contains two double values: x1 y1 and the second one also contains two double values with one digit after the decimal point: x2 y2.

**Output**
Calculate and print the distance value using the provided formula, with 4 decimal places.

|Input Sample|Output Sample|
|-|-|
|1.0 7.0|4.4721|
|5.0 9.0||
|||
|-2.5 0.4|16.1484|
|12.1 7.3||
|||
|2.5 -0.4|16.4575|
|-12.2 7.0||


In [ ]:
point_1 = input().split()
point_2 = input().split()
point_1 = [float(point) for point in point_1]
point_2 = [float(point) for point in point_2]

distance = (((point_1[0]-point_2[0])**2)+((point_1[1]-point_2[1]))**2)**(1/2)

print(f'{distance:.4f}')

2.5 -0.4
-12.2 7.0
16.4575


#####BEE 1016 -	Distance


Two cars (X and Y) leave in the same direction. The car X leaves with a constant speed of 60 km/h and the car Y leaves with a constant speed of 90 km / h.

In one hour (60 minutes) the car Y can get a distance of 30 kilometers from the X car, in other words, it can get away one kilometer for each 2 minutes.

Read the distance (in km) and calculate how long it takes (in minutes) for the car Y to take this distance in relation to the other car.

**Input**
The input file contains 1 integer value.

**Output**
Print the necessary time followed by the message "minutos" that means minutes in Portuguese.

|Input Sample|Output Sample|
|-|-|
|30|60 minutos|
|110|220 minutos|
|7|14 minutos|

In [ ]:
distance = int(input())
time = distance*2
print(f'{time} minutos')

7
14 minutos


#####BEE 1017 - Fuel Spent

Little John wants to calculate and show the amount of spent fuel liters on a trip, using a car that does 12 Km/L. For this, he would like you to help him through a simple program. To perform the calculation, you have to read spent time (in hours) and the same average speed (km/h). In this way, you can get distance and then, calculate how many liters would be needed. Show the value with three decimal places after the point.

**Input**
The input file contains two integers. The first one is the spent time in the trip (in hours). The second one is the average speed during the trip (in Km/h).

**Output**
Print how many liters would be needed to do this trip, with three digits after the decimal point.

|Input Sample|Output Sample|
|-|-|
|10|70.833|
|85||
|||
|2|15.333|
|92||
|||
|22|122.833|
|67||

In [ ]:
time = int(input()) #h
avg_speed = int(input()) #km/h
CAR_EFFICIENCY = 12 #km/l
consumption = time*avg_speed/CAR_EFFICIENCY

print(f'{consumption:.3f}')




10
85
70.833


#####BEE 1018 - Banknotes

In this problem you have to read an integer value and calculate the smallest possible number of banknotes in which the value may be decomposed. The possible banknotes are 100, 50, 20, 10, 5, 2 and 1. Print the read value and the list of banknotes.

Input
The input file contains an integer value N (0 < N < 1000000).

Output
Print the read number and the minimum quantity of each necessary banknotes in Portuguese language, as the given example. Do not forget to print the end of line after each line, otherwise you will receive “Presentation Error”.

|Input Sample|Output Sample|
|-|-|
|576|576|
||5 nota(s) de R\$ 100.00|
||1 nota(s) de R\$ 50.00|
||1 nota(s) de R\$ 20.00|
||0 nota(s) de R\$ 10.00|
||1 nota(s) de R\$ 5.00|
||0 nota(s) de R\$ 2.00|
||1 nota(s) de R\$ 1.00|


In [ ]:
cash = int(input())
banknotes = 0
print(cash)
banknotes = cash // 100
cash = cash % 100
print(f'{banknotes} nota(s) de R$ 100,00')
banknotes = cash // 50
cash = cash % 50
print(f'{banknotes} nota(s) de R$ 50,00')
banknotes = cash // 20
cash = cash % 20
print(f'{banknotes} nota(s) de R$ 20,00')
banknotes = cash // 10
cash = cash % 10
print(f'{banknotes} nota(s) de R$ 10,00')
banknotes = cash // 5
cash = cash % 5
print(f'{banknotes} nota(s) de R$ 5,00')
banknotes = cash // 2
cash = cash % 2
print(f'{banknotes} nota(s) de R$ 2,00')
banknotes = cash // 1
cash = cash % 1
print(f'{banknotes} nota(s) de R$ 1,00')


299
299
2 nota (s) de R$ 100,00
1 nota (s) de R$ 50,00
2 nota (s) de R$ 20,00
0 nota (s) de R$ 10,00
1 nota (s) de R$ 5,00
2 nota (s) de R$ 2,00
0 nota (s) de R$ 1,00


#####BEE 1019 - Time Conversion

Read an integer value, which is the duration in seconds of a certain event in a factory, and inform it expressed in hours:minutes:seconds.

Input
The input file contains an integer N.

Output
Print the read time in the input file (seconds) converted in hours:minutes:seconds like the following example.

|Input Sample|Output Sample|
|-|-|
|556|0:9:16|
|||
|1|0:0:1|
|||
|140153|38:55:53|


In [ ]:
time = int(input())
hour = time // 3600
time = time % 3600
minute = time // 60
second = time % 60

print(f'{hour}:{minute}:{second}')

7000
1:56:40


#####BEE 1020 - Age in Days

Read an integer value corresponding to a person's age (in days) and print it in years, months and days, followed by its respective message “ano(s)”, “mes(es)”, “dia(s)”.

Note: only to facilitate the calculation, consider the whole year with 365 days and 30 days every month. In the cases of test there will never be a situation that allows 12 months and some days, like 360, 363 or 364. This is just an exercise for the purpose of testing simple mathematical reasoning.

**Input**
The input file contains 1 integer value.

**Output**
Print the output, like the following example.

|Input Sample|Output Sample|
|-|-|
|400|1 ano(s)|
||1 mes(es)|
||5 dia(as)|
|||
|800|2 ano(s)|
||2 mes(es)|
||10 dia(as)|
|||
|30|0 ano(s)|
||1 mes(es)|
||0 dia(as)|

In [ ]:
dia = int(input())
ano = dia // 365
dia = dia - (365*ano)
mes = dia // 30
dia = dia - (30*mes)
print(f'{ano} ano(s)\n{mes} mes(es)\n{dia} dia(s)')

400
1 ano(s)
1 mes(es)
5 dia(s)


#####BEE 1021 - Banknotes and Coins

Read a value of floating point with two decimal places. This represents a monetary value. After this, calculate the smallest possible number of notes and coins on which the value can be decomposed. The considered notes are of 100, 50, 20, 10, 5, 2. The possible coins are of 1, 0.50, 0.25, 0.10, 0.05 and 0.01. Print the message “NOTAS:” followed by the list of notes and the message “MOEDAS:” followed by the list of coins.

**Input**
The input file contains a value of floating point N (0 ≤ N ≤ 1000000.00).

**Output**
Print the minimum quantity of banknotes and coins necessary to change the initial value, as the given example.

|Input Sample|Output Sample|
|-|-|
|576.73|NOTAS:|
||5 nota(s) de R\$ 100.00|
||1 nota(s) de R\$ 500.00|
||1 nota(s) de R\$ 20.00|
||0 nota(s) de R\$ 10.00|
||1 nota(s) de R\$ 5.00|
||0 nota(s) de R\$ 2.00|
||MOEDAS:|
||1 moeda(s) de R\$ 1.00|
||1 moeda(s) de R\$ 0.50|
||0 moeda(s) de R\$ 0.25|
||2 moeda(s) de R\$ 0.10|
||0 moeda(s) de R\$ 0.05|
||3 moeda(s) de R\$ 0.01|

In [ ]:
value = float(input())
notes = [2.00, 5.00, 10.00, 20.00, 50.00, 100.00]
coins = [ 1.00, 0.50, 0.25, 0.10, 0.05, 0.01]
notes = notes[::-1]
qt_notes = dict()
for x in notes:
  v = value//x
  value = value - (v * x)
  qt_notes.update({x:v})
qt_coins = dict()
for x in coins:
  v = value//x
  value = value - (v*x)
  qt_coins.update({x:v})

print('NOTAS:')

for x, y in qt_notes.items():
  print(f'{y:.0f} nota(s) de R$ {x:.2f}')
print('MOEDAS:')

for x, y in qt_coins.items():
  print(f'{y:.0f} moeda(s) de R$ {x:.2f}')


91.01
NOTAS:
0 nota(s) de R$ 100.00
1 nota(s) de R$ 50.00
2 nota(s) de R$ 20.00
0 nota(s) de R$ 10.00
0 nota(s) de R$ 5.00
0 nota(s) de R$ 2.00
MOEDAS:
1 moeda(s) de R$ 1.00
0 moeda(s) de R$ 0.50
0 moeda(s) de R$ 0.25
0 moeda(s) de R$ 0.10
0 moeda(s) de R$ 0.05
1 moeda(s) de R$ 0.01


In [ ]:

value = float(input())*100
_notes = [2.00, 5.00, 10.00, 20.00, 50.00, 100.00]
notes = [x*100 for x in _notes]
_coins = [ 1.00, 0.50, 0.25, 0.10, 0.05, 0.01]
coins = [x*100 for x in _coins]
notes = notes[::-1]
qt_notes = dict()
for x in notes:
  v = value//x
  value = value - (v * x)
  qt_notes.update({x:v})
qt_coins = dict()
for x in coins:
  v = value//x
  value = value - (v*x)
  qt_coins.update({x:v})

print('NOTAS:')

for x, y in qt_notes.items():
  print(f'{y:.0f} nota(s) de R$ {x/100:.2f}')
print('MOEDAS:')

for x, y in qt_coins.items():
  print(f'{y:.0f} moeda(s) de R$ {x/100:.2f}')

400
NOTAS:
4 nota(s) de R$ 100.00
0 nota(s) de R$ 50.00
0 nota(s) de R$ 20.00
0 nota(s) de R$ 10.00
0 nota(s) de R$ 5.00
0 nota(s) de R$ 2.00
MOEDAS:
0 moeda(s) de R$ 1.00
0 moeda(s) de R$ 0.50
0 moeda(s) de R$ 0.25
0 moeda(s) de R$ 0.10
0 moeda(s) de R$ 0.05
0 moeda(s) de R$ 0.01


#####BEE 1035 - Selection Test 1

Read 4 integer values A, B, C and D. Then if B is greater than C and D is greater than A and if the sum of C and D is greater than the sum of A and B and if C and D were positives values and if A is even, write the message “Valores aceitos” (Accepted values). Otherwise, write the message “Valores nao aceitos” (Values not accepted).

**Input**
Four integer numbers A, B, C and D.

**Output**
Show the corresponding message after the validation of the values​​.

|Input Sample|Output Sample|
|-|-|
|5 6 7 8|Valores nao aceitos|
|||
|2 3 2 6|Valores aceitos|



In [ ]:
#1035
list = input().split()
size = len(list)
list[0] = int(list[0])
list = [int(list[x]) for x in range(0,size)]
condition1 = list[1]>list[2] and list[3]>list[0]
condition2 = list[2]+list[3] > list[0]+list[1]
condition3 = list[2]>0 and list[3]>0 and list[0]%2==0
condition = condition1 and condition2 and condition3
if condition:
  print('Valores aceitos')
else:
  print('Valores nao aceitos')

2 3 2 6
Valores aceitos


#####BEE 1036 - Bhaskara's Formula

Read 3 floating-point numbers. After, print the roots of bhaskara’s formula. If it's impossible to calculate the roots because a division by zero or a square root of a negative number, presents the message “Impossivel calcular”.

**Input**
Read 3 floating-point numbers (double) A, B and C.

**Output**
Print the result with 5 digits after the decimal point or the message if it is impossible to calculate.

|Input Samples|Output Samples|
|-|-|
|10.0 20.1 5.1|R1 = -0.29788|
||R2 = -1.71212|
|||
|0.0 20.0 5.0|Impossivel calcular|
|||
|10.3 203.0 5.0|R1 = -0.02466|
||R2 = -19.68408|
|||
|10.0 3.0 5.0|Impossivel calcular|

In [ ]:
#1036

list = input().split()
list = [float(list[x]) for x in range(0,len(list))]
delta = (list[1]**2)-(4*list[0]*list[2])

if list[0]==0 or delta<0:
  print('Impossivel calcular')
else:
  r1 = (-list[1]+(delta**(1/2)))/(2*list[0])
  r2 = (-list[1]-(delta**(1/2)))/(2*list[0])
  print(f'R1 = {r1:.5f}\nR2 = {r2:.5f}')

10.0 3.0 5.0
-191.0
Impossivel calcular


#####BEE 1037 - Interval

You must make a program that read a float-point number and print a message saying in which of following intervals the number belongs: [0,25] (25,50], (50,75], (75,100]. If the read number is less than zero or greather than 100, the program must print the message “Fora de intervalo” that means "Out of Interval".

The symbol '(' represents greather than. For example:
[0,25] indicates numbers between 0 and 25.0000, including both.
(25,50] indicates numbers greather than 25 (25.00001) up to 50.0000000.

**Input**
The input file contains a floating-point number.

**Output**
The output must be a message like following example.

|Input Sample|Output Sample|
|-|-|
|25.01|Intervalo (25,50]|
|||
|25.00|Intervalo [0,25]|
|||
|100.00|Intervalo (75,100]|
|||
|-25.02|Fora de intervalo|

In [ ]:
number = float(input())
if number < 0 or number > 100:
  print('Fora de intervalo')
elif number >= 0 and number <=25:
  print('Intervalo [0,25]')
elif number > 25 and number <=50:
  print('Intervalo (25,50]')
elif number > 50 and number <=75:
  print('Intervalo (50,76]')
elif number > 75 and number <=100:
  print('Intervalo (75,100]')

100.00
Intervalo (75,100]


#####BEE 1038 - Snack

Using the following table, write a program that reads a code and the amount of an item. After, print the value to pay. This is a very simple program with the only intention of practice of selection commands.

|CODE|SPECIFICATION|PRICE|
|-|-|-|
|1|Cachorro Quente|R\$ 4.00|
|2|X-Salda|R\$ 4.50|
|3|X-Bacon|R\$ 5.00|
|4|Torrada simples|R\$ 2.00|
|5|Refrigerante|R\$ 1.50|

**Input**
The input file contains two integer numbers X and Y. X is the product code and Y is the quantity of this item according to the above table.

**Output**
The output must be a message "Total: R$ " followed by the total value to be paid, with 2 digits after the decimal point.

|Input Sample|Output Sample|
|-|-|
|3 2|Total: R\$ 10.00|
|||
|4 3|Total: R\$ 6.00|
|||
|2 3|Total: R\$ 13.50|
|||

In [ ]:
dict = {
  1:('Cachorro Quente',4.00),
  2:('X-salada',4.50),
  3:('X-Bacon',5.00),
  4:('Torrada simples',2.00),
  5:('Refrigerante',1.50)
}
list = input().split()
list = [int(x) for x in list]
print(f'Total: R$ {(dict[list[0]][1])*list[1]:.2f}')

1 6
Total: R$ 24.00


#####BEE 1040 - Average 3

Read four numbers (N1, N2, N3, N4), which one with 1 digit after the decimal point, corresponding to 4 scores obtained by a student. Calculate the average with weights 2, 3, 4 e 1 respectively, for these 4 scores and print the message "Media: " (Average), followed by the calculated result. If the average was 7.0 or more, print the message "Aluno aprovado." (Approved Student). If the average was less than 5.0, print the message: "Aluno reprovado." (Reproved Student). If the average was between 5.0 and 6.9, including these, the program must print the message "Aluno em exame." (In exam student).

In case of exam, read one more score. Print the message "Nota do exame: " (Exam score) followed by the typed score. Recalculate the average (sum the exam score with the previous calculated average and divide by 2) and print the message “Aluno aprovado.” (Approved student) in case of average 5.0 or more) or "Aluno reprovado." (Reproved student) in case of average 4.9 or less. For these 2 cases (approved or reproved after the exam) print the message "Media final: " (Final average) followed by the final average for this student in the last line.

**Input**
The input contains four floating point numbers that represent the students' grades.

**Output**
Print all the answers with one digit after the decimal point.

|Input Sample|Output Sample|
|-|-|
|2.0 4.0 7.5 8.0|Media: 5.4|
|6.4|Aluno em exame.|
||Nota do exame: 6.4|
||Aluno aprovado.|
||Media final: 5.9|
|||
|2.0 6.5 4.0 9.0|Media: 4.8|
||Aluno reprovado.|
|||
|9.0 4.0 8.5 9.0|Media: 7.3|
||Aluno aprovado.|


In [ ]:
def average():
  list = input().split()
  list = [float(x) for x in list]
  avg = ((list[0]*2)+(list[1]*3)+(list[2]*4)+(list[3]*1))/(2+3+4+1)
  return avg
avg = average()

print(f'Media: {avg:.1f}')
if avg>=7.0:
  status = 'Aluno aprovado.'
  print(f'{status}')
elif avg<5.0:
  status = 'Aluno reprovado.'
  print(f'{status}')
elif avg>=5.0 and avg<7.0:
  status = 'Aluno em exame.'
  print(f'{status}')
if status == 'Aluno em exame.':
  score_5 = float(input())
  avg_final = (avg+score_5)/2
  print(f'Nota do exame: {score_5}')
  if avg_final>=5.0:
    print('Aluno aprovado.')
  else:
    print('Aluno reprovado.')
  print(f'Media final: {avg_final:.1f}')


2 4 7.5 8
Média: 5.4
Aluno em exame.
6.4
Nota do exame: 6.4
Aluno aprovado.
Media final: 5.9


#####BEE 1041 - Coordinates of a Point

Write an algorithm that reads two floating values (x and y), which should represent the coordinates of a point in a plane. Next, determine which quadrant the point belongs, or if you are at one of the Cartesian axes or the origin (x = y = 0).

<img src=https://resources.beecrowd.com/gallery/images/problems/UOJ_1041.png>

If the point is at the origin, write the message "Origem".

If the point is at X axis write "Eixo X", else if the point is at Y axis write "Eixo Y".

**Input**
The input contains the coordinates of a point.

**Output**
The output should display the quadrant in which the point is.

|Input Sample|Output Sample|
|-|-|
|4.5 -2.2|Q4|
|||
|0.1 0.1|Q1|
|||
|0.0 0.0|Origem|

In [ ]:
point = input().split()
point = [float(x) for x in point]
if point[0]==0 and point[1]==0:
  print('Origem')
elif point[0]==0 and point[1]!=0:
  print('Eixo Y')
elif point[1]==0 and point[0]!=0:
  print('Eixo X')
elif point[0]>0 and point[1]>0:
  print('Q1')
elif point[0]<0 and point[1]>0:
  print('Q2')
elif point[0]<0 and point[1]<0:
  print('Q3')
elif point[0]>0 and point[1]<0:
  print('Q4')


-45 -15
Q3


#####1042 Simple Sort

Read three integers and sort them in ascending order. After, print these values in ascending order, a blank line and then the values in the sequence as they were readed.

Input
The input contains three integer numbers.

Output
Present the output as requested above.

|INPUT SAMPLE|OUTPUT SAMPLE|
|------------|-------------|
|7 21 -14|-14|
||7|
||21|
|||
||-14|
||7|
||21|
|||

|INPUT SAMPLE|OUTPUT SAMPLE|
|------------|-------------|
|-14 21 7|-14|
||7|
||21|
|||
||-14|
||21|
||7|



In [ ]:
list = input().split()
list = [int(x) for x in list]
list_read = list[::]
list.sort()
for x in list:
  print(x)
print('')
for x in list_read:
  print(x)

-14 9 2
-14
2
9

-14
9
2


#####BEE 1043 - Triangle

Read three point floating values (A, B and C) and verify if is possible to make a triangle with them. If it is possible, calculate the perimeter of the triangle and print the message:


Perimetro = XX.X


If it is not possible, calculate the area of the trapezium which basis A and B and C as height, and print the message:


Area = XX.X

Input
The input file has tree floating point numbers.

Output
Print the result with one digit after the decimal point.

|INPUT SAMPLE|OUTPUT SAMPLE|
|------------|-------------|
|6.0 4.0 2.0|Area = 10.0|
|6.0 4.0 2.1|Perimetro = 12.1|




In [ ]:
sizes = input().split()
sizes = [float(x) for x in sizes]
condition = sizes[0]+sizes[1]>sizes[2] and sizes[1]+sizes[2]>sizes[0] and sizes[2]+sizes[0]>sizes[1]
if condition:
  per = sum(sizes)
  print(f'Perimetro = {per:.1f}')
else:
  area = ((sizes[0]+sizes[1])*sizes[2])/2
  print(f'Area = {area:.1f}')

 6.0 4.0 2.1
Perimetro = 12.10


#####BEE 1044 - Multiples

Read two integer values (A and B). After, the program should print the message "Sao Multiplos" (are multiples) or "Nao sao Multiplos" (aren’t multiples), corresponding to the read values.

Input
The input has two integer numbers.

Output
Print the relative message to the input as stated above.

|Input Sample|Output Sample|
|------------|-------------|
|6 24        |Sao Multiplos|
|6 25        |Nao sao Multiplos|


In [ ]:
values = input().split()
values = [int(x) for x in values]
a = values[0]
b = values[1]

if a>b:
  if a%b==0:
    print("Sao Multiplos")
  else:
    print('Nao sao Multiplos')
elif b>a:
  if b%a==0:
    print("Sao Multiplos")
  else:
    print('Nao sao Multiplos')
elif a==b:
  print('Sao Multiplos')

24 6
Sao Multiplos


#####BEE 1045 - Triangle Types

Read 3 double numbers (A, B and C) representing the sides of a triangle and arrange them in decreasing order, so that the side A is the biggest of the three sides. Next, determine the type of triangle that they can make, based on the following cases always writing an appropriate message:

*  if A ≥ B + C, write the message: NAO FORMA TRIANGULO
*  if A² = B² + C², write the message: TRIANGULO RETANGULO
*  if A² > B² + C², write the message: TRIANGULO OBTUSANGULO
*  if A² < B² + C², write the message: TRIANGULO ACUTANGULO
*  if the three sides are the same size, write the message: TRIANGULO EQUILATERO
*  if only two sides are the same and the third one is different, write the message: TRIANGULO ISOSCELES

**Input**

The input contains three double numbers, A (0 < A) , B (0 < B) and C (0 < C).

**Output**

Print all the classifications of the triangle presented in the input.

|Input Samples|Output Samples|
|-------------|--------------|
|7.0 5.0 7.0|TRIANGULO ACUTANGULO|
| |TRIANGULO ISOSCELES|
|6.0 6.0 10.0|TRIANGULO OBTUSANGULO|
| |TRIANGULO ISOSCELES|
|6.0 6.0 6.0|TRIANGULO ACUTANGULO|
| |TRIANGULO EQUILATERO|
|5.0 7.0 2.0|NAO FORMA TRIANGULO|


In [ ]:
sides = [float(x) for x in input().split()]
sides.sort(reverse=True)
A, B, C = sides
if A>=(B+C):
  print('NAO FORMA TRIANGULO')
elif A**2 == ((B**2)+(C**2)):
  print('TRIANGULO RETANGULO')
elif A**2 > ((B**2)+(C**2)):
  print('TRIANGULO OBTUSANGULO')
elif A**2 < ((B**2)+(C**2)):
  print('TRIANGULO ACUTANGULO')
if A==B and B==C:
  print('TRIANGULO EQUILATERO')
if (A==B and B!=C) or (A==C and A!=C) or (B==C and B!=A):
  print('TRIANGULO ISOSCELES')

6 6 10
TRIANGULO OBTUSANGULO
TRIANGULO ISOSCELES


#####BEE 1046 - Game Time


Read the start time and end time of a game, in hours. Then calculate the duration of the game, knowing that the game can begin in a day and finish in another day, with a maximum duration of 24 hours. The message must be printed in portuguese “O JOGO DUROU X HORA(S)” that means “THE GAME LASTED X HOUR(S)”

**Input**

Two integer numbers representing the start and end time of a game.

**Output**

Print the duration of the game as in the sample output.

|Input Sample|Output Sample|
|------------|-------------|
|16 2|O JOGO DUROU 10 HORA(S)|
|0 0|O JOGO DUROU 24 HORA(S)|
|2 16|O JOGO DUROU 14 HORA(S)|


In [ ]:
time = [int(x) for x in input().split()]
start = time[0]
end = time[1]
counter = 0
#print(start, end, counter)
if start<end:
  counter = end - start
else:
  while start<24:
    #print(start, counter)
    counter+=1
    start+=1
    if start == 24:

      start = 0
      while start<end:
        #print(start, counter)
        counter+=1
        start+=1
        #if counter == 24:
         # break
      break

print(f'O JOGO DUROU {counter:.0f} HORA(S)')

2 16
O JOGO DUROU 14 HORA(S)


#####BEE 1047 - Game Time with Minutes

Read the start time and end time of a game, in hours and minutes (initial hour, initial minute, final hour, final minute). Then print the duration of the game, knowing that the game can begin in a day and finish in another day,

Obs.: With a maximum game time of 24 hours and the minimum game time of 1 minute.

**Input**
Four integer numbers representing the start and end time of the game.

**Output**
Print the duration of the game in hours and minutes, in this format: “O JOGO DUROU XXX HORA(S) E YYY MINUTO(S)” . Which means: the game lasted XXX hour(s) and YYY minutes.

|INPUT SAMPLE|OUTPUT SAMPLE|
|------------|-------------|
|7 8 9 10|O JOGO DUROU 2 HORA(S) E 2 MINUTO(S)|
|7 7 7 7|O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)|
|7 10 8 9|O JOGO DUROU 0 HORA(S) E 59 MINUTO(S)|




In [ ]:
lista = list((map(int, input().split())))
start = (lista[0]*3600) + (lista[1]*60)
end = (lista[2]*3600) + (lista[3]*60)
temp = 0
seconds = 0
#print(start, end)
#if start == end:
#  hours = 24
#  minutes = 0
#  print(f'O JOGO DUROU {hours} HORA(S) E {minutes} MINUTO(S)')
if end>start:
    seconds = end-start
else:
    count = 0
    while start<86400:
      #print(f'start {start}, count {count}')
      count+=1
      start+=1
    if start == 86400:
      start = 0
      while start<end:
        #print(f'start {start}, count {count}')
        start+=1
        count+=1
    seconds = count
#print(seconds)
hours = seconds//3600
seconds = seconds%3600
minutes = seconds//60
print(f'O JOGO DUROU {hours} HORA(S) E {minutes} MINUTO(S)')

20 45 3 15
O JOGO DUROU 6 HORA(S) E 30 MINUTO(S)


#####BEE 1048 - Salary Increase

The company ABC decided to give a salary increase to its employees, according to the following table:

|Salary|Readjustment Rate|
|-|-|
|0 - 400.00|15%|
|400.01 - 800.00|12%|
|800.01 - 1200.00|10%|
|1200.01 - 2000.00|7%|
|Above 2000.00|4%|

Read the employee's salary, calculate and print the new employee's salary, as well the money earned and the increase percentual obtained by the employee, with corresponding messages in Portuguese, as the below example.

**Input**
The input contains only a floating-point number, with 2 digits after the decimal point.

**Output**
Print 3 messages followed by the corresponding numbers (see example) informing the new salary, the among of money earned (both must be shown with 2 decimal places) and the percentual obtained by the employee. Note:
Novo salario:  means "New Salary"
Reajuste ganho: means "Money earned"
Em percentual: means "In percentage"

|Input Sample|Output Sample|
|-|-|
|400.00|Novo salario: 460.00|
||Reajuste ganho: 60.00|
||Em percentual: 15 %|
|||
|800.01|Novo salario: 880.01|
||Reajuste ganho: 80.00|
||Em percentual: 10 %|
|||
|2000.00|Novo salario: 2140.00|
||Reajuste ganho: 140.00|
||Em percentual: 7 %|

In [ ]:
def adjustment(salary):
  if salary<=400:
    percent = 15
    adjustment = salary*percent/100
    new_salary = salary+adjustment
  elif salary<=800:
    percent = 12
    adjustment = salary*percent/100
    new_salary = salary+adjustment
  elif salary<=1200:
    percent = 10
    adjustment = salary*percent/100
    new_salary = salary+adjustment
  elif salary<=2000:
    percent = 7
    adjustment = salary*percent/100
    new_salary = salary+adjustment
  else:
    percent = 4
    adjustment = salary*percent/100
    new_salary = salary+adjustment
  return print(f'Novo salario: {new_salary:.2f}\nReajuste ganho: {adjustment:.2f}\nEm percentual: {percent} %')

adjustment(float(input()))

2000
Novo salario: 2140.00
Reajuste ganho: 140.00
Em percentual: 7 %


#####BEE 1049 - Animal

In this problem, your job is to read three Portuguese words. These words define an animal according to the table below, from left to right. After, print the chosen animal defined by these three words.

<img src=https://resources.beecrowd.com/gallery/images/problems/UOJ_1049_b.png>

**Input**
The input contains 3 words, one by line, that will be used to identify the animal, according to the above table, with all letters in lowercase.

**Output**
Print the animal name according to the given input.

|Input Sample|Output Sample|
|-|-|
|vertebrado|homem|
|mamifero||
|onivoro||
|||
|vertebrado|aguia|
|ave||
|carnivoro||
|||
|invertebrado|minhoca|
|anelideo||
|onivoro||

In [6]:
word_1 = input()
word_2 = input()
word_3 = input()

if word_1 == 'vertebrado':
  if word_2 == 'ave':
    if word_3 == 'carnivoro':
      print('aguia')
    elif word_3 == 'onivoro':
      print('pomba')
  elif word_2 == 'mamifero':
    if word_3 == 'onivoro':
      print('homem')
    elif word_3 == 'herbivoro':
      print('vaca')
elif word_1 == 'invertebrado':
  if word_2 == 'inseto':
    if word_3 == 'hematofago':
      print('pulga')
    elif word_3 == 'herbivoro':
      print('lagarta')
  elif word_2 == 'anelideo':
    if word_3 == 'hematofago':
      print('sanguessuga')
    elif word_3 == 'onivoro':
      print('minhoca')

invertebrado
anelideo
onivoro
minhoca


#####BEE 1050 - DDD

Read an integer number that is the code number for phone dialing. Then, print the destination according to the following table:

<img src=https://resources.beecrowd.com/gallery/images/problems/UOJ_1050.png>

If the input number isn’t found in the above table, the output must be:
DDD nao cadastrado
That means “DDD not found” in Portuguese language.

**Input**
The input consists in a unique integer number.

**Output**
Print the city name corresponding to the input DDD. Print DDD nao cadastrado if doesn't exist corresponding DDD to the typed number.

|Input Sample|Output Sample|
|-|-|
|11|Sao Paulo|

In [12]:
DDD = {
    61:'Brasilia',
    71:'Salvador',
    11:'Sao Paulo',
    21:'Rio de Janeiro',
    32:'Juiz de Fora',
    19:'Campinas',
    27:'Vitoria',
    31:'Belo Horizonte'
}
ddd_input = int(input())
if ddd_input in DDD:
  print(DDD[ddd_input])
else:
  print('DDD nao cadastrado')

28
DDD nao cadastrado


###BEE 1051 - Taxes

In an imaginary country called Lisarb, all the people are very happy to pay their taxes because they know that doesn’t exist corrupt politicians and the taxes are used to benefit the population, without any misappropriation. The currency of this country is Rombus, whose symbol is R$.

Read a value with 2 digits after the decimal point, equivalent to the salary of a Lisarb inhabitant. Then print the due value that this person must pay of taxes, according to the table below.

<img src=https://resources.beecrowd.com/gallery/images/problems/UOJ_1051_en.png>

Remember, if the salary is R$ 3,002.00 for example, the rate of 8% is only over R$ 1,000.00, because the salary from R$ 0.00 to R$ 2,000.00 is tax free. In the follow example, the total rate is 8% over R$ 1000.00 + 18% over R$ 2.00, resulting in R$ 80.36 at all. The answer must be printed with 2 digits after the decimal point.

**Input**
The input contains only a float-point number, with 2 digits after the decimal point.

**Output**
Print the message "R$" followed by a blank space and the total tax to be payed, with two digits after the decimal point. If the value is up to 2000, print the message "Isento".

|Input Sample|Output Sample|
|-|-|
|3002.00|R\$ 80.36|
|1701.12|Isento|
|4520.00|R\$ 355.60|
